Added basic CSS, navigation bar, js refactoring, and examples

Prospero23 opened the pull request:
Hey Rodrigo, 
Here is my attempt at adding a basic version of the interaction example I posted in vexflow/vexflow#265. 

While I was doing that, I thought it would be nice to also add basic styling in line with https://www.vexflow.com and also just refactor out the showHtml logic. 

I changed the examples directory to "demos" as that seems potentially more accurate to the contents, but I am more than happy to change it back if you disagree. 

Finally, I added a separate directory for each of the  demos/examples as it seems easier to search by url and allows each page to be more focused on a single demo. 

Edit: I had some time the past few days, so decided to take a swing at adding all the @ronyeh examples that are in the main repo as I saw this was discussed in #2 a while ago. 

Thanks,
Ben

Prospero23 commented:
I am also down to change the css to use something like bulma or bootstrap if that would be better.

ronyeh commented:
Hi Ben!

Thanks for joining the team. ;-)

I haven't had a chance to review this yet, but some quick notes:
- Can we try to keep most file names in lowercase? interaction-SVG => interaction-svg? I know we have README.md and Gruntfile.js and LICENSE but those are special cases.
- @rvilarl I'm happy to move the demos away from the main repo into this one if you all think this is the better way forward. However, I think there is a benefit to having *some* basic examples in the same main vexflow repo so that when you clone and build the project for the first time you have something easy to verify that everything's working well.

Also, as we move forward we need to consider that many devs will use AI/LLM assistants from now on. Having solid examples in the main repo will help the AI IDEs generate correct code.

Prospero23 commented:
@ronyeh keeping some demos on the main repo makes complete sense to me. I don't see why all the examples that are already there couldn't stay. I will obviously default to whatever you guys think is best, as I am new around here :)

Prospero23 commented:
node demos handling leaves much to be desired, but I will tackle that in next few days when I have time. Right now I just have tags linking to the js files but it would be easy enough to create actual pages for them.

ronyeh commented:
I'll wait a bit to give Rodrigo a chance to chime in.

But in general it makes sense for *most* of the examples to go in this repository.

Maybe some "getting started" demos and bare-bones sanity check demos can stay in the main repo. And the full suite of tests, of course.

We can also include a README.md to direct developers to this repo with all the examples.

rvilarl commented:
> I'll wait a bit to give Rodrigo a chance to chime in.
> 
> But in general it makes sense for _most_ of the examples to go in this repository.
> 
> Maybe some "getting started" demos and bare-bones sanity check demos can stay in the main repo. And the full suite of tests, of course.
> 
> We can also include a README.md to direct developers to this repo with all the examples.

@ronyeh, I agree with your proposal above.
@Prospero23, Thanks for going on with your collaboration! Nothing against the renaming to demos and directory approach. There are also tutorial steps from v4 missing in this repo, if you want to go further see https://github.com/0xfe/vexflow/wiki/Tutorial

Prospero23 commented:
@rvilarl I will check out adding those. Just made some fixes that should make all the js and css actually work with github pages. 

I didn't realize how pages makes absolute paths weird, so I have just removed the navigation I made as it relied on that and changed all the js and css links to be relative. 

It may be nice to use some sort of build process that can make having static urls easier but this works for now

Prospero23 commented:
https://prospero23.github.io/vexflow-examples/ 

hosted version if you want to see what it currently looks like

rvilarl commented:
Looks very good. It would be nice to standardize the look and feel of the demos providing code and jsfiddle, would you agree? We do not have to do that here but step by step in future PRs

Prospero23 commented:
Yeah, I completely  agree. 

With that in mind, would it make sense to transition to  using some sort of either templating engine or static site generator? I assume the less boilerplate people deal with having to make, the easier contributing will be if people want to get involved, and the more likely that styling will remain consistent.

Using your `showHtml` function for those examples is already kind of doing this (and may be enough), but with switching to some sort of formal templating library or static site generator it may make this process slightly more streamlined without having to reinvent a lot of the functionality that already exists in many libraries/frameworks. I know that with using an SSG, stuff like searching can be built in, which would obviously be pretty nice to not have to maintain as examples/tutorials grow.

Using something like mustache.js for basic templating could even work, but I am just thinking about what the easiest way to scale will be as we hopefully add more examples and try to enforce standardization. 

That being said, if we don't want to do that, I can also just work towards making the examples look/feel more standardized in future PRs. 

Let me know what you think,
Ben

ronyeh commented:
I've been experimenting with 11ty as a static site generator. We should probably decide as a team to pick one and go with it, if it saves us time in generating these HTML docs & example pages.

For this PR, just do it the "hard way" without another tool.

When you're ready to go, let us know and we can review & merge. Then we can explore how tools like 11ty or Jekyll or others can help us.

ronyeh commented:
> While I was doing that, I thought it would be nice to also add basic styling in line with https://www.vexflow.com and also just refactor out the showHtml logic.

Where's `showHtml()`?

FYI, vexflow.com hasn't been updated for a while now. That's one of our TODOs. Once we get V5 super solid, we should update the main website with Mohit's help.

So I wouldn't worry about matching the styling on vexflow.com. Maybe you can pick a different  them that is usable and responsive and modern (in a separate PR with a static site generator). Then if the new theme works for everyone, maybe vexflow.com will follow your lead! :-)

Prospero23 commented:
@ronyeh main.js has the function definition for `showHtml` but it is used in the two original examples (hairpin and textbracket) along with the interactivity/interaction demo I added. 

I agree on saving all the static site stuff for another day. I believe I can finish up this PR tomorrow. All I want to do is add the last few steps of the tutorial from V4 and do some cleanup from the aftermath of me removing navigation.

Prospero23 commented:
@ronyeh @rvilarl Everything should be ready for review/merging now.

I think finishing out the tutorial can be a separate/smaller PR.

rvilarl commented:
Thanks Ben! I think that the best is to merge and then resolve the issues if any. Let us improve this examples now little by little.
